### PR TITLE
fix(experiment-core): match array user property values element-wise for non-set operators

### PR DESCRIPTION
**Pull request by kyeh-amp**
## Summary
- Non-set operators (`is`, `contains`, `greater`, etc.) now match array user property values element-wise (any-match semantics) instead of stringifying the whole array, aligning with Amplitude analytics/charts behavior.
- Fixes a latent bug in `coerceStringArray` where `value` (not `parsedValue`) was iterated after a successful `JSON.parse`, and adds a `startsWith("[")` pre-check to skip the JSON parse for scalars.
- Switches the evaluation integration test deployment key to the VVhLULX superset deployment and adds coverage for array properties (both native arrays and JSON array strings) across `is`, `is not`, `contains`, `does not contain`, and `set is`.

## Test plan
- [x] `yarn build` passes (5 projects)
- [x] `yarn test` passes (141 tests across 8 suites in `experiment-core`, plus all other packages)
- [x] New unit tests in `packages/experiment-core/test/evaluation/evaluation-array-values.test.ts` (14 cases covering scalars, native arrays, JSON array strings, malformed JSON, empty arrays, leading whitespace)
- [x] 7 new integration tests inline with existing operator tests in `evaluation-integration.test.ts`

## Behavioral change
Consumers relying on the previous stringified-array behavior for non-set operators (e.g., matching `'["a","b"]'` literally against `'["a","b"]'` with `is`) will see different results. This is intentional — the new behavior matches the Amplitude analytics/charts contract.

[SKY-10281]: https://amplitude.atlassian.net/browse/SKY-10281?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

> [!NOTE]
> **Medium Risk**
> Changes core flag/segment condition matching semantics for array-valued user properties, which can alter targeting outcomes for existing flags; mitigated by added unit/integration test coverage.
> 
> **Overview**
> Updates `EvaluationEngine.matchCondition` so **non-set operators** (e.g. `is`, `contains`, comparisons) treat array-valued properties (native arrays or JSON array strings) as *any-match element-wise* instead of matching against a stringified array.
> 
> Fixes `coerceStringArray` parsing to only attempt JSON parsing for strings that start with `[` and to iterate over the parsed array result, and adds new unit/integration tests covering array behavior and edge cases; the integration test deployment key is also updated to a deployment that includes the new flag fixtures.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit baab8336969e3937f6c0b5117d5f1924c380b617. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/experiment-core/src/evaluation/evaluation.ts [hunk 1]
```diff
@@ -411,6 +420,16 @@ export class EvaluationEngine {
...
 
+  private matchStringsNonSet(
+    propValues: string[],
+    op: string,
+    filterValues: string[],
+  ): boolean {
+    return propValues.some((element) =>
+      this.matchString(element, op, filterValues),
+    );
+  }
```
[the remark this answers is not in the document]
Comment (kyeh-amp) [reply packages/experiment-core/src/evaluation/evaluation.ts:431]: This is consistent with evaluation-core and mirrors the observed behavior in chart filtering.

**Review by vaibhav-jain-exp — APPROVED** [review]
